### ISSUE TITLE
Valid Loss Increasing While Train Loss Decreases Using Custom Training Data with Speaker ID

### ISSUE THREAD

**Issue by zjttfs** [speechbrain/speechbrain#2584]
### Describe the bug

I have 463 speakers, and for each speaker, I have prepared 140 audio files for training, 30 for testing, and 30 for validation. Each audio file is approximately 10-120 seconds long. After multiple attempts, the train loss continuously decreases, but the validation loss is slowly increasing, and the validation error remains at the upper limit of 2.0e-02. How should I address this issue?

### Expected behaviour

that both the train loss and the validation loss would decrease over time as the model learns from the training data.

### To Reproduce

train.yaml
https://drive.google.com/file/d/1Fo8-hovjfmKTX3SfUJV5bZ1JcYIjuqr_/view?usp=sharing
train.[y
https://drive.google.com/file/d/1H3PsL2jQcYz5s7kW-uC0Egum2_0zTc_t/view?usp=sharing

### Environment Details

```
audioread                3.0.1     
certifi                  2024.2.2  
cffi                     1.16.0    
charset-normalizer       3.3.2     
decorator                5.1.1     
filelock                 3.13.3    
fsspec                   2024.3.1  
huggingface-hub          0.22.2    
HyperPyYAML              1.2.2     
idna                     3.6       
Jinja2                   3.1.3     
joblib                   1.3.2     
lazy-loader              0.4       
librosa                  0.10.1    
llvmlite                 0.42.0    
MarkupSafe               2.1.5     
mpmath                   1.3.0     
msgpack                  1.0.8     
networkx                 3.2.1     
numba                    0.59.1    
numpy                    1.26.4    
nvidia-cublas-cu12       12.1.3.1  
nvidia-cuda-cupti-cu12   12.1.105  
nvidia-cuda-nvrtc-cu12   12.1.105  
nvidia-cuda-runtime-cu12 12.1.105  
nvidia-cudnn-cu12        8.9.2.26  
nvidia-cufft-cu12        11.0.2.54 
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.19.3    
nvidia-nvjitlink-cu12    12.4.99   
nvidia-nvtx-cu12         12.1.105  
packaging                24.0      
pillow                   10.3.0    
pip                      20.0.2    
pkg-resources            0.0.0     
platformdirs             4.2.0     
pooch                    1.8.1     
pycparser                2.22      
PyYAML                   6.0.1     
requests                 2.31.0    
ruamel.yaml              0.18.6    
ruamel.yaml.clib         0.2.8     
scikit-learn             1.4.2     
scipy                    1.12.0    
sentencepiece            0.2.0     
setuptools               44.0.0    
soundfile                0.12.1    
soxr                     0.3.7     
speechbrain              1.0.0     
sympy                    1.12      
threadpoolctl            3.4.0     
torch                    2.2.2     
torchaudio               2.2.2     
torchvision              0.17.2    
tqdm                     4.66.2    
triton                   2.2.0     
typing-extensions        4.10.0    
urllib3                  2.2.1  
```

### Relevant Log Output

```shell
Epoch: 1, lr: 1.00e-03 - train loss: 8.84e-01 - valid loss: 1.76e-01, valid error: 5.25e-02
Epoch: 2, lr: 9.95e-04 - train loss: 2.20e-01 - valid loss: 1.21e-01, valid error: 3.72e-02
Epoch: 3, lr: 9.91e-04 - train loss: 1.65e-01 - valid loss: 8.99e-02, valid error: 3.00e-02
Epoch: 4, lr: 9.86e-04 - train loss: 1.30e-01 - valid loss: 8.33e-02, valid error: 3.00e-02
Epoch: 5, lr: 9.82e-04 - train loss: 1.21e-01 - valid loss: 6.92e-02, valid error: 2.53e-02
Epoch: 6, lr: 9.77e-04 - train loss: 1.06e-01 - valid loss: 8.20e-02, valid error: 2.95e-02
Epoch: 7, lr: 9.73e-04 - train loss: 9.26e-02 - valid loss: 6.69e-02, valid error: 2.50e-02
Epoch: 8, lr: 9.68e-04 - train loss: 8.52e-02 - valid loss: 6.61e-02, valid error: 2.38e-02
Epoch: 9, lr: 9.64e-04 - train loss: 7.97e-02 - valid loss: 6.60e-02, valid error: 2.44e-02
Epoch: 10, lr: 9.59e-04 - train loss: 7.65e-02 - valid loss: 6.02e-02, valid error: 2.29e-02
Epoch: 11, lr: 9.55e-04 - train loss: 7.63e-02 - valid loss: 5.45e-02, valid error: 2.24e-02
Epoch: 12, lr: 9.50e-04 - train loss: 7.14e-02 - valid loss: 6.25e-02, valid error: 2.47e-02
Epoch: 13, lr: 9.46e-04 - train loss: 6.31e-02 - valid loss: 5.86e-02, valid error: 2.42e-02
Epoch: 14, lr: 9.41e-04 - train loss: 6.28e-02 - valid loss: 5.85e-02, valid error: 2.18e-02
Epoch: 15, lr: 9.37e-04 - train loss: 6.04e-02 - valid loss: 6.12e-02, valid error: 2.09e-02
Epoch: 16, lr: 9.32e-04 - train loss: 5.87e-02 - valid loss: 6.05e-02, valid error: 2.30e-02
Epoch: 17, lr: 9.28e-04 - train loss: 5.77e-02 - valid loss: 6.06e-02, valid error: 2.31e-02
Epoch: 18, lr: 9.23e-04 - train loss: 5.40e-02 - valid loss: 5.55e-02, valid error: 2.10e-02
Epoch: 19, lr: 9.19e-04 - train loss: 5.17e-02 - valid loss: 5.42e-02, valid error: 2.19e-02
Epoch: 20, lr: 9.14e-04 - train loss: 5.42e-02 - valid loss: 5.41e-02, valid error: 2.09e-02
Epoch: 21, lr: 9.10e-04 - train loss: 4.87e-02 - valid loss: 5.84e-02, valid error: 2.16e-02
Epoch: 22, lr: 9.05e-04 - train loss: 5.24e-02 - valid loss: 6.18e-02, valid error: 2.13e-02
Epoch: 23, lr: 9.01e-04 - train loss: 4.61e-02 - valid loss: 5.43e-02, valid error: 2.15e-02
Epoch: 24, lr: 8.96e-04 - train loss: 4.59e-02 - valid loss: 5.48e-02, valid error: 2.04e-02
Epoch: 25, lr: 8.91e-04 - train loss: 4.56e-02 - valid loss: 5.34e-02, valid error: 2.15e-02
Epoch: 26, lr: 8.87e-04 - train loss: 4.30e-02 - valid loss: 5.19e-02, valid error: 2.06e-02
Epoch: 27, lr: 8.82e-04 - train loss: 4.43e-02 - valid loss: 5.57e-02, valid error: 2.17e-02
Epoch: 28, lr: 8.78e-04 - train loss: 4.35e-02 - valid loss: 4.87e-02, valid error: 1.99e-02
Epoch: 29, lr: 8.73e-04 - train loss: 4.24e-02 - valid loss: 5.10e-02, valid error: 2.02e-02
Epoch: 30, lr: 8.69e-04 - train loss: 4.21e-02 - valid loss: 5.53e-02, valid error: 1.99e-02
Epoch: 31, lr: 8.64e-04 - train loss: 3.86e-02 - valid loss: 4.39e-02, valid error: 2.04e-02
Epoch: 32, lr: 8.60e-04 - train loss: 4.38e-02 - valid loss: 6.63e-02, valid error: 1.99e-02
Epoch: 33, lr: 8.55e-04 - train loss: 3.79e-02 - valid loss: 6.73e-02, valid error: 2.10e-02
Epoch: 34, lr: 8.51e-04 - train loss: 3.70e-02 - valid loss: 6.80e-02, valid error: 2.09e-02
Epoch: 35, lr: 8.46e-04 - train loss: 3.75e-02 - valid loss: 7.07e-02, valid error: 2.03e-02
Epoch: 36, lr: 8.42e-04 - train loss: 3.58e-02 - valid loss: 6.00e-02, valid error: 2.03e-02
Epoch: 37, lr: 8.37e-04 - train loss: 3.50e-02 - valid loss: 7.36e-02, valid error: 2.05e-02
Epoch: 38, lr: 8.33e-04 - train loss: 3.32e-02 - valid loss: 7.17e-02, valid error: 2.13e-02
Epoch: 39, lr: 8.28e-04 - train loss: 3.35e-02 - valid loss: 7.73e-02, valid error: 2.06e-02
Epoch: 40, lr: 8.24e-04 - train loss: 3.18e-02 - valid loss: 6.66e-02, valid error: 2.01e-02
Epoch: 41, lr: 8.19e-04 - train loss: 3.12e-02 - valid loss: 6.73e-02, valid error: 1.98e-02
Epoch: 42, lr: 8.15e-04 - train loss: 3.30e-02 - valid loss: 7.26e-02, valid error: 1.97e-02
Epoch: 43, lr: 8.10e-04 - train loss: 2.88e-02 - valid loss: 7.44e-02, valid error: 2.00e-02
Epoch: 44, lr: 8.06e-04 - train loss: 3.11e-02 - valid loss: 7.58e-02, valid error: 1.92e-02
Epoch: 45, lr: 8.01e-04 - train loss: 3.12e-02 - valid loss: 7.23e-02, valid error: 2.03e-02
Epoch: 46, lr: 7.96e-04 - train loss: 2.91e-02 - valid loss: 8.54e-02, valid error: 2.12e-02
Epoch: 47, lr: 7.92e-04 - train loss: 2.73e-02 - valid loss: 7.20e-02, valid error: 1.94e-02
Epoch: 48, lr: 7.87e-04 - train loss: 2.73e-02 - valid loss: 7.44e-02, valid error: 2.04e-02
Epoch: 49, lr: 7.83e-04 - train loss: 2.60e-02 - valid loss: 7.50e-02, valid error: 2.01e-02
Epoch: 50, lr: 7.78e-04 - train loss: 2.62e-02 - valid loss: 8.19e-02, valid error: 1.91e-02
Epoch: 51, lr: 7.74e-04 - train loss: 2.67e-02 - valid loss: 9.04e-02, valid error: 2.05e-02
Epoch: 52, lr: 7.69e-04 - train loss: 2.52e-02 - valid loss: 8.12e-02, valid error: 2.03e-02
Epoch: 53, lr: 7.65e-04 - train loss: 2.73e-02 - valid loss: 8.45e-02, valid error: 2.01e-02
Epoch: 54, lr: 7.60e-04 - train loss: 2.88e-02 - valid loss: 8.65e-02, valid error: 2.02e-02
Epoch: 55, lr: 7.56e-04 - train loss: 2.80e-02 - valid loss: 8.91e-02, valid error: 2.01e-02
Epoch: 56, lr: 7.51e-04 - train loss: 2.43e-02 - valid loss: 7.88e-02, valid error: 2.03e-02
Epoch: 57, lr: 7.47e-04 - train loss: 2.64e-02 - valid loss: 8.00e-02, valid error: 2.07e-02
Epoch: 58, lr: 7.42e-04 - train loss: 2.43e-02 - valid loss: 7.58e-02, valid error: 2.04e-02
Epoch: 59, lr: 7.38e-04 - train loss: 2.24e-02 - valid loss: 7.35e-02, valid error: 1.93e-02
Epoch: 60, lr: 7.33e-04 - train loss: 2.44e-02 - valid loss: 8.97e-02, valid error: 2.01e-02
Epoch: 61, lr: 7.29e-04 - train loss: 2.31e-02 - valid loss: 7.88e-02, valid error: 1.99e-02
Epoch: 62, lr: 7.24e-04 - train loss: 2.50e-02 - valid loss: 8.88e-02, valid error: 2.04e-02
Epoch: 63, lr: 7.20e-04 - train loss: 2.41e-02 - valid loss: 8.02e-02, valid error: 2.04e-02
Epoch: 64, lr: 7.15e-04 - train loss: 2.64e-02 - valid loss: 8.17e-02, valid error: 1.97e-02
Epoch: 65, lr: 7.11e-04 - train loss: 2.33e-02 - valid loss: 7.85e-02, valid error: 2.01e-02
Epoch: 66, lr: 7.06e-04 - train loss: 2.00e-02 - valid loss: 8.46e-02, valid error: 2.02e-02
Epoch: 67, lr: 7.02e-04 - train loss: 2.25e-02 - valid loss: 7.54e-02, valid error: 1.94e-02
```
```


### Additional Context

_No response_
